ResearchKit/Common/ORKOrderedTask.m

```diff
@@ -1446,15 +1439,118 @@ + (ORKOrderedTask *)tremorTestTaskWithIdentifier:(NSString *)identifier
+            step.text = intendedUseDescription ? : ORKLocalizedString(@"TREMOR_TEST_INTENDED_USE", nil);
+            step.image = [UIImage imageNamed:@"tremortest1" inBundle:[NSBundle bundleForClass:[self class]] compatibleWithTraitCollection:nil];
+            step.shouldTintImages = YES;
+            
+            ORKStepArrayAddStep(steps, step);
+        }
+        {
+            ORKInstructionStep *step = [[ORKInstructionStep alloc] initWithIdentifier:ORKInstruction1StepIdentifier];
+            step.title = ORKLocalizedString(@"TREMOR_TEST_TITLE", nil);
+            
+            if (handOptions == ORKPredefinedTaskHandOptionUndefined) {
+                step.text = ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DEFAULT_TEXT", nil);
+            } else {
+                if (firstIsLeft) {
+                    step.text = ORKLocalizedString(@"TREMOR_TEST_INTRO_2_LEFT_HAND_TEXT", nil);
+                } else {
+                    step.text = ORKLocalizedString(@"TREMOR_TEST_INTRO_2_RIGHT_HAND_TEXT", nil);
+                }
+            }
+            
+            NSArray<NSString *>*detailStringForNumberOfTasks = @[
+                                                                 ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_1_TASK", nil),
+                                                                 ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_2_TASK", nil),
+                                                                 ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_3_TASK", nil),
+                                                                 ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_4_TASK", nil),
+                                                                 ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_5_TASK", nil)
+                                                                 ];
+
+            // start with the count for all the tasks, then subtract one for each excluded task flag
+            static const NSInteger allTasks = 5; // hold in lap, outstretched arm, elbow bent, repeatedly touching nose, queen wave
+            NSInteger actualTasksIndex = allTasks - 1;
+            NSInteger bits = sizeof(activeTaskOptions) * CHAR_BIT; // don't assume exclude flags are sequential--check all positions
+            for (NSInteger i = 0; i < bits ; ++i) {
+                if (activeTaskOptions & (1 << i)) {
+                    actualTasksIndex--;
+                }
+            }
+            
+            NSString *detailFormat = (handOptions & ORKPredefinedTaskHandOptionBoth) == ORKPredefinedTaskHandOptionBoth ? ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_BOTH_HANDS_%", nil) : ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_DEFAULT_%", nil);
+            step.detailText = [NSString stringWithFormat:detailFormat, detailStringForNumberOfTasks[actualTasksIndex]];
+            step.image = [UIImage imageNamed:@"tremortest2" inBundle:[NSBundle bundleForClass:[self class]] compatibleWithTraitCollection:nil];
```

Comment:
WRT the code, it's in AppCore

On Tue, Mar 8, 2016 at 9:50 PM, Erin-Mounts <notifications@github.com>
wrote:

> In ResearchKit/Common/ORKOrderedTask.m
> <https://github.com/Sage-Bionetworks/ResearchKit/pull/17#discussion_r55474494>
> :
>
> > +                                                                 ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_5_TASK", nil)
> > +                                                                 ];
> > +
> > +            // start with the count for all the tasks, then subtract one for each excluded task flag
> > +            static const NSInteger allTasks = 5; // hold in lap, outstretched arm, elbow bent, repeatedly touching nose, queen wave
> > +            NSInteger actualTasksIndex = allTasks - 1;
> > +            NSInteger bits = sizeof(activeTaskOptions) * CHAR_BIT; // don't assume exclude flags are sequential--check all positions
> > +            for (NSInteger i = 0; i < bits ; ++i) {
> > +                if (activeTaskOptions & (1 << i)) {
> > +                    actualTasksIndex--;
> > +                }
> > +            }
> > +
> > +            NSString *detailFormat = (handOptions & ORKPredefinedTaskHandOptionBoth) == ORKPredefinedTaskHandOptionBoth ? ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_BOTH_HANDS_%", nil) : ORKLocalizedString(@"TREMOR_TEST_INTRO_2_DETAIL_DEFAULT_%", nil);
> > +            step.detailText = [NSString stringWithFormat:detailFormat, detailStringForNumberOfTasks[actualTasksIndex]];
> > +            step.image = [UIImage imageNamed:@"tremortest2" inBundle:[NSBundle bundleForClass:[self class]] compatibleWithTraitCollection:nil];
>
> actually I was planning to ask Ian for a left-handed image (since the
> phone itself isn't symmetrical) and put that in tomorrow. but does this
> flippedImage method already exist somewhere in mPower or would I need to
> add it to a UIImage category?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Sage-Bionetworks/ResearchKit/pull/17/files#r55474494>.
>



-- 
*Shannon Young* | Senior Software Engineer |
Sage Bionetworks | 1100 Fairview Ave N, M1-C108 | Seattle, WA 98109 |
c: 206-335-8922 <206.335.8922> | e: shannon.young@sagebase.org